crates/router/src/core/payment_link.rs

```diff
@@ -93,80 +129,80 @@ The get_js_script function is used to inject dynamic value to payment_link sdk,
-        "window.__PAYMENT_DETAILS_STR = JSON.stringify({{
-        client_secret: '{secret}',
-        amount: '{amount}',
-        currency: '{currency}',
-        payment_id: '{payment_id}',
-        expiry: {expiry},
-        // TODO: Remove hardcoded values
-        merchant_logo: 'https://upload.wikimedia.org/wikipedia/commons/8/83/Steam_icon_logo.svg',
-        return_url: 'http://localhost:5500/public/index.html',
-        currency_symbol: '$',
-        merchant: 'Steam',
-        max_items_visible_after_collapse: 3,
-        order_details: [
-            {{
-              product_name:
-                'dskjghbdsiuh sagfvbsajd ugbfiusedg fiudshgiu sdhgvishd givuhdsifu gnb gidsug biuesbdg iubsedg bsduxbg jhdxbgv jdskfbgi sdfgibuh ew87t54378 ghdfjbv jfdhgvb dufhvbfidu hg5784ghdfbjnk f (taxes incl.)',
-              quantity: 2,
-              amount: 100,
-              product_image:
-                'https://upload.wikimedia.org/wikipedia/commons/8/83/Steam_icon_logo.svg',
-            }},
-            {{
-              product_name: \"F1 '23\",
-              quantity: 4,
-              amount: 500,
-              product_image:
-                'https://upload.wikimedia.org/wikipedia/commons/8/83/Steam_icon_logo.svg',
-            }},
-            {{
-              product_name: \"Motosport '24\",
-              quantity: 4,
-              amount: 500,
-              product_image:
-                'https://upload.wikimedia.org/wikipedia/commons/8/83/Steam_icon_logo.svg',
-            }},
-            {{
-              product_name: 'Trackmania',
-              quantity: 4,
-              amount: 500,
-              product_image:
-                'https://upload.wikimedia.org/wikipedia/commons/8/83/Steam_icon_logo.svg',
-            }},
-            {{
-              product_name: 'Ghost Recon',
-              quantity: 4,
-              amount: 500,
-              product_image:
-                'https://upload.wikimedia.org/wikipedia/commons/8/83/Steam_icon_logo.svg',
-            }},
-            {{
-              product_name: 'Cup of Tea',
-              quantity: 4,
-              amount: 500,
-              product_image:
-                'https://upload.wikimedia.org/wikipedia/commons/8/83/Steam_icon_logo.svg',
-            }},
-            {{
-              product_name: 'Tea cups',
-              quantity: 4,
-              amount: 500,
-              product_image:
-                'https://upload.wikimedia.org/wikipedia/commons/8/83/Steam_icon_logo.svg',
-            }},
-          ]
-    }});
-
-    const hyper = Hyper(\"{pub_key}\");"
+        ":root {{
+      --primary-color: {primary_color};
+      --primary-accent-color: {primary_accent_color};
+      --secondary-color: {secondary_color};
+    }}"
     )
 }
+
+fn validate_sdk_requirements(
+    pub_key: Option<String>,
+    currency: Option<api_models::enums::Currency>,
+    client_secret: Option<String>,
+    return_url: (Option<String>, Option<String>),
```

Comment:
take these are two different fields, one as `merchant_account_return_url` and the other as `payment_return_url`, this way it is clear which url to pass in which field when calling the function. The problem in this case is that the caller has to look into the function definition as to know which url to pass where as both are option of string

Reply:
You can also calculate the return url before calling this function